File: backend/src/main/java/com/zhilv/dto/TripRequest.java
package com.zhilv.dto;

import java.math.BigDecimal;
import java.util.List;

/**
 * 表单式旅行规划请求(POST /api/trips 入参)。
 * 只有用户填的表单条件(目的地/日期/预算/偏好),不含每日安排 days——
 * 每日安排要等生成链路(D13+)跑完,放进 TripPlan 的 plan_json。
 * 字段名 snake_case,与前端表单字段一致。
 */
public record TripRequest(
        String title,
        String destination,
        String departure,               // 出发城市(每单必填)
        String start_date,
        String end_date,
        Integer travelers,
        BigDecimal budget,
        List<String> preferences,
        String pace,
        String hotel_level,              // 舒适型 / 高档型 / 经济型
        List<String> dietary_preferences, // 如 ["少辣"]
        String special_notes
) {}

File: ai-service/app/models/trip_request.py
"""生成请求入参模型(字段与 Java com.zhilv.dto.TripRequest 对齐,全部 snake_case)。"""

from __future__ import annotations

from datetime import date
from typing import Optional

from pydantic import BaseModel, Field, field_validator, model_validator


class TripRequest(BaseModel):
    title: Optional[str] = None
    destination: str
    departure: Optional[str] = None     # 出发城市(每单必填;旧请求无则 None)
    start_date: str                     # yyyy-MM-dd,与 Java LocalDate.parse 同格式
    end_date: str                       # yyyy-MM-dd
    travelers: int = Field(default=2, ge=1)
    budget: float = Field(gt=0)
    preferences: list[str] = Field(default_factory=list)
    pace: Optional[str] = None          # 轻松 / 适中 / 紧凑
    hotel_level: Optional[str] = None   # 舒适型 / 高档型 / 经济型
    dietary_preferences: Optional[list[str]] = None   # 如 ["少辣"]
    special_notes: Optional[str] = None

    @field_validator("start_date", "end_date")
    @classmethod
    def _validate_date_format(cls, v: str) -> str:
        try:
            date.fromisoformat(v)
        except ValueError:
            raise ValueError("日期格式须为 yyyy-MM-dd,如 2026-09-01")
        return v

    @model_validator(mode="after")
    def _end_not_before_start(self) -> "TripRequest":
        if date.fromisoformat(self.end_date) < date.fromisoformat(self.start_date):
            raise ValueError("结束日期不能早于开始日期")
        return self
